accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,290 @@
+
+**User story 2**
+
+User Story: As an engineer responsible for maintaining the Uyuni codebase, I want to efficiently investigate and fix reported bugs or performance issues within the Uyuni infrastructure. To expedite the debugging process and ensure a stable codebase, I would like to leverage the capabilities of the integrated health check tool, which will help me identify any underlying health-related problems that could be contributing to the reported bug.
+
+As an engineer, I get the supportconfig mostly so I need this tool to consume the provided supportconfig and help me identify any potential issues before I dive into the huge amount of logs.
+
+
+## Expected outcome
+
+* Proactive Monitoring and Maintenance: Enable system administrators to proactively monitor the health and performance of the Uyuni infrastructure. By providing comprehensive insights into the system's status, the tool aims to help administrators detect and resolve potential issues before they escalate into significant problems.
+
+* Streamlined Debugging Process for engineers: Assist engineers and developers in quickly identifying and addressing bugs or performance issues within the Uyuni codebase.
+
+# Detailed design
+
+[design]: #detailed-design
+
+
+## Component Overview
+
+### Health-Check-Tool
+
+* Loki/Promtail: These two components are for log aggregation, indexing and querying. Loki acts as the centralized logging system, storing and managing logs, while Promtail is deployed on the Uyuni server to collect logs and forward them to Loki.
+
+* Grafana: Utilized for visualization, with dashboards displaying metrics, logs, and alerts, facilitating an integrated view of system health and performance.
+
+* Uyuni-health-exporter: A custom exporter to gather specific metrics related to Uyuni server health, including database, service status, and resource usage.
+
+* Alerting System: Leveraging Grafana's alerting mechanism, configured to monitor critical metrics and log patterns, triggering alerts based on predefined thresholds.
+
+* LogCLI: A command-line interface tool provided by Grafana Loki, LogCLI enables direct querying of logs stored in Loki from the terminal. It supports running LogQL queries, live tailing of logs, and viewing logs in a flexible output format. This tool is particularly useful for administrators and developers who prefer to work within a CLI environment for quick log searches, debugging, and log analysis without the need for a graphical interface.
+
+URL: https://github.com/uyuni-project/poc-uyuni-health-check
+
+### Saline
+
+Saline is an addition for Salt used in SUSE Manager/Uyuni aimed to provide better control and visibility for states deployment in the large scale environments. It can provide metrics related to Salt events and "state.apply" process on the minions.
+
+URL: https://github.com/vzhestkov/saline
+
+## Configuration and Deployment
```

Comment:
This section lists possible implementation options whereas we should name here which solutions we decide for. E.g. we could write:

- integrated solution: extending existing monitoring stack
- standalone aka disconnected solution: containerized deployment including only Loki, Promtail and Grafana.

Reply:
I've updated the text, so it is more clear now which are the  proposed solutions.